js/controllers/grid/navigationMenus/form/NavigationMenuFormHandler.js

```diff
@@ -0,0 +1,271 @@
+								)
+							);
+						}
+					}
+				}
+		);
+
+		// Reset any nesting that's been carried over from the assigned list
+		$('#pkpNavUnassigned > li').each(
+				function() {
+					var $childList = $(this).children('ul');
+					if ($childList.length) {
+						$childList.find('li').each(function() {
+							$(this).appendTo($('#pkpNavUnassigned'));
+						});
+					}
+					$childList.remove();
+				}
+		);
+
+		// Initialize the sortable controls
+		$('#pkpNavManagement ul').sortable({
+			placeholder: 'pkp_nav_item_placeholder',
+			delay: 250,
+			connectWith: '#pkpNavManagement ul',
+			update: this.callbackWrapper(this.updateSorting),
+			start: function() {
+				$('#pkpNavAssigned').addClass('pkp_is_sorting');
+			},
+			stop: function() {
+				$('#pkpNavAssigned').removeClass('pkp_is_sorting');
+			}
+		});
+	};
+
+
+	/**
+	 * Re-initialize the .sortable() components and generate the form fields
+	 * whenever the list is re-sorted.
+	 */
+	$.pkp.controllers.grid.navigationMenus.form.NavigationMenuFormHandler
+			.prototype.updateSorting = function() {
+		var $navManagement = $('#pkpNavManagement'),
+				seq = 0,
+				parent = null,
+				currentName = '';
+
+		// Re-intialize the sortable component after adjusting sort order
+		this.initSorting();
+
+		// Remove existing hidden fields
+		$('input', $navManagement).remove();
+
+		// Generate new hidden form fields
+		$('#pkpNavAssigned > li').each(function() {
+			currentName = 'menuTree[' + $(this).data('id') + ']';
+			$navManagement.append('<input type="hidden" name="' +
+					currentName + '[seq]" value="' +
+					seq + '">');
+			seq++;
+
+			var parentId = $(this).data('id');
+			$(this).find('li').each(function() {
+				currentName = 'menuTree[' + $(this).data('id') + ']';
+				$navManagement.append('<input type="hidden" name="' +
+						currentName + '[seq]" value="' +
+						seq + '">');
+				$navManagement.append('<input type="hidden" name="' +
+						currentName + '[parentId]" value="' +
+						parentId + '">');
+				seq++;
+			});
+		});
+	};
+
+
+	/**
+	 * Show the conditional display warning message
+	 */
+	$.pkp.controllers.grid.navigationMenus.form.NavigationMenuFormHandler
+			.prototype.showConditionalDisplayWarning = function(htmlElement) {
+		var itemType = $(htmlElement).closest('li').data('type');
+
+		if (this.itemTypeConditionalWarnings_[itemType] !== null) {
+			var opts = {
+				title: this.warningModalTitle_,
+				okButton: this.okButton_,
+				cancelButton: false,
+				dialogText: this.itemTypeConditionalWarnings_[itemType]
+			};
+
+			$('<div id="' + $.pkp.classes.Helper.uuid() + '" ' +
+					'class="pkp_modal pkpModalWrapper" tabindex="-1"></div>')
+				.pkpHandler('$.pkp.controllers.modal.ConfirmationModalHandler', opts);
+		}
+
+		return false;
+	}
+
+
+	/**
+	 * Show the submenu link warning message
+	 */
+	$.pkp.controllers.grid.navigationMenus.form.NavigationMenuFormHandler
+			.prototype.showSubmenuWarning = function(e) {
```

Comment:
`e` is unused and can be removed. (I know, this is my code... :) )

Reply:
OK